tests/python_/typesets/test_python_complete_set.py

```diff
@@ -0,0 +1,459 @@
+    "str_complex_nan": Complex,
+    "uuid_series": UUID,
+    "uuid_series_str": UUID,
+    "uuid_series_missing": UUID,
+    "ip_mixed_v4andv6": IPAddress,
+    "file_test_py": File,
+    "file_test_py_missing": File,
+    "file_mixed_ext": File,
+    "image_png": Image,
+    "image_png_missing": Image,
+    "str_int_leading_zeros": String,
+    "str_float_non_leading_zeros": Float,
+    "str_int_zeros": Integer,
+    "email_address": EmailAddress,
+    "email_address_missing": EmailAddress,
+    "email_address_str": EmailAddress,
+    "py_datetime_str": DateTime,
+    "all_null_nan": Generic,
+    "all_null_nat": Generic,
+    "all_null_none": Generic,
+    "all_null_empty_str": String,
+}
+if pandas_version[0] >= 1:
+    inference_map["string_dtype_series"] = String
+
+
+@pytest.mark.parametrize(**get_inference_cases(series, inference_map, typeset))
+def test_inference(name, series, inference_type, typeset, difference):
+    """Test the generated combinations for "inference(series) == type"
+
+    Args:
+        series: the series to test
+        inference_type: the type to test against
+    """
+    result, message = infers(name, series, inference_type, typeset, difference)
+    assert result, message
+
+
+# Conversions in one single step
+convert_map = [
+    # Model type, Relation type
+    (Integer, Float, {"int_nan_series", "float_series2"}),
+    (Complex, String, {"str_complex", "str_complex_nan"}),
+    (
+        Float,
+        String,
+        {
+            "string_flt",
+            "string_num_nan",
+            "string_num",
+            "string_flt_nan",
+            "textual_float",
+            "textual_float_nan",
+            "int_str_range",
+            "str_float_non_leading_zeros",
+            "str_int_zeros",
+            # "string_with_sep_num_nan",
+        },
+    ),
+    (Date, DateTime, {"date_series_nat", "datetime"}),
+    (DateTime, String, {"timestamp_string_series", "string_date", "py_datetime_str"}),
+    (Geometry, String, {"geometry_string_series"}),
+    (Boolean, String, {"string_bool_nan"}),
+    (IPAddress, String, {"ip_str"}),
+    (URL, String, {"str_url"}),
+    (Path, String, {"path_series_windows_str", "path_series_linux_str"}),
+    (EmailAddress, String, {"email_address_str"}),
+    (Float, Complex, {"complex_series_float", "complex_series_py_float"}),
+    (Boolean, Object, {"bool_nan_series", "mixed"}),
+    (UUID, String, {"uuid_series_str"}),
+]
+
+
+@pytest.mark.parametrize(**get_convert_cases(series, convert_map, typeset))
+def test_conversion(name, source_type, relation_type, series, member):
+    """Test the generated combinations for "convert(series) == type" and "infer(series) = source_type"
+
+    Args:
+        series: the series to test
+        type: the type to test against
+    """
+    result, message = convert(name, source_type, relation_type, series, member)
+    assert result, message
+
+
+cast_results = {
+    "float_series2": pd.Series([1, 2, 3, 4], dtype=np.int64),
```

Comment:
Shouldn't the series be python sequences here?

Reply:
We do value comparisons for all of the tests rather than object comparisons so the tests weren't affected. I pushed an update removing the Series objects though!